mrmap/registry/tests/acceptance/steps/steps.py

```diff
@@ -20,3 +23,116 @@ def step_impl(context, json_path):
     results = jsonpath(json_body, json_path)
     if results:
         fail('Match found at <{path}> for <{body}>'.format(path=json_path, body=json_body))
+
+
+@then('all {the_class} objects are returned')
+def get_all_resources(context, the_class):
+    if the_class == 'services':
+        total_number_of_features = Service.objects.count()
+    elif the_class == 'layers':
+        total_number_of_features = Layer.objects.count()
+    elif the_class == 'featuretypes':
+        total_number_of_features = FeatureType.objects.count()
+    elif the_class == 'dataset_metadata':
+        total_number_of_features = DatasetMetadata.objects.count()
+    else:
+        total_number_of_features = 0
+
+    response = context.response
+    results = response.json()
+
+    # the response is actually paginated, and if more than 20 results are in total number of features,
+    # then len(results['results']) will be 20
+    if int(results['count']) > 20:
+        assert_that(len(results['results'])).is_equal_to(20)
+        assert_that(int(results['total_pages'])).is_equal_to(math.ceil(int(total_number_of_features) / len(results['results'])))
+    else:
+        assert_that(len(results['results'])).is_equal_to(int(total_number_of_features))
+    assert_that(int(results['count'])).is_equal_to(int(total_number_of_features))
+
+
+# TODO: refactor to be only one step that makes the call and gives the response, instead of giving the url in a
```

Comment:
the  idea  behind  it  has also to  do with the comment you left next.

Right now, we are using the sentences provided by `behave-restful` ((e.g. `Given a request URL http://localhost:8000/api/v1/registry/service/services/?search=WMS&ordering=title`.

Then we are checking that we are sending a get, and then the response status is OK

In my opinion, these steps are a bit redundant and can be condensed in one step.

Furthermore, instead of providing a request URL like the one being  provided right now, where the parameters are visible, I think it would be better to have these parameters defined in a sentence.

To have a general view of what I mean, I think the tests would be much cleaner by being able to write:

```
Given a **GET** request to URL **http://localhost:8000/api/v1/registry/service/services/**
Then **1** result(s) is(are) returned, with **WMS** as **search** criteria
And response objects are ordered **ascending** by **title**
```

instead of

```
Given a request url **http://localhost:8000/api/v1/registry/service/services/?search=WMS&ordering=title**
When the request sends **GET**
Then the response status is **OK**
And **1** result(s) is(are) returned, with **WMS** as search **criteria**
And response objects are ordered **ascending** by **title**
```

That is why the parameters in the functions are unused. When I wrote them, I had this idea in mind. Unfortunately it is not possible to do this with `behave-restful` and we need to develop our own methods. Hence the TODO

Hope this explanation was clear enough :)

Reply:
Yes, thank you. Understood and makes sense to me :-)